cachi2/interface/cli.py

```diff
@@ -0,0 +1,96 @@
+import importlib.metadata
+import json
+import logging
+from itertools import chain
+from pathlib import Path
+from typing import Optional, Union
+
+import typer
+from typer import Option
+
+from cachi2.core.models import Request
+from cachi2.core.package_managers import gomod
+
+app = typer.Typer()
+log = logging.getLogger(__name__)
+
+DEFAULT_SOURCE = "."
+DEFAULT_OUTPUT = "./cachi2-output"
+
+
+def version_callback(value: bool) -> None:
+    """If --version was used, print the cachi2 version and exit."""
+    if value:
+        print("cachi2", importlib.metadata.version("cachi2"))
+        raise typer.Exit()
+
+
+@app.callback()
+def cachi2(  # noqa: D103; docstring becomes part of --help message
+    version: bool = Option(
+        False,
+        "--version",
+        callback=version_callback,
+        is_eager=True,
+        help="Show version and exit.",
+    ),
+) -> None:
+    # Process top-level options here
+    pass
+
+
+def maybe_load_json(opt_name: str, opt_value: str) -> Optional[Union[dict, list]]:
+    """If the option string looks like a JSON dict or list, parse it. Otherwise, return None."""
+    if not opt_value.lstrip().startswith(("{", "[")):
+        return None
+
+    try:
+        value = json.loads(opt_value)
+    except json.JSONDecodeError:
+        raise typer.BadParameter(f"{opt_name}: looks like JSON but is not valid JSON")
+
+    return value
+
+
+@app.command()
+def fetch_deps(
+    package: list[str] = Option(
+        ...,  # Ellipsis makes this option required
+        help="Specify package (within the source repo) to process. Can be used multiple times.",
+        metavar="PKG",
+    ),
+    source: Path = Option(DEFAULT_SOURCE, help="Process the git repository at this path."),
+    output: Path = Option(DEFAULT_OUTPUT, help="Write output files to this directory."),
+    # TODO: let's have actual flags like --gomod-vendor instead?
+    flags: str = Option(
+        "",
+        help="Pass additional flags as a comma-separated list.",
+        metavar="FLAGS",
+    ),
```

Comment:
WDYT? Having `--flags=flag1,flag2,flag3` as opposed to `--flag1 --flag2 --flag3` feels very anti-CLI

Reply:
+1 for a single `--flags=str[]`  flag :)

Reply:
+1 for `cachi2 fetch-deps --gomod-vendor --cgo-disable`

Reply:
Added a WIP commit that allows both forms, LMK what you think